collections/map.go

```diff
@@ -119,14 +126,14 @@ func (m Map[K, V]) getStore(ctx context.Context) (storetypes.KVStore, error) {
 	return provider.KVStore(m.sk), nil
 }
 
-func (m Map[K, V]) encodeKey(key K) ([]byte, error) {
-	prefixLen := len(m.prefix)
+func encodeKeyWithPrefix[K any](prefix []byte, kc KeyCodec[K], key K) ([]byte, error) {
```

Comment:
```suggestion
func encodeKeyForCollection[K any](collectionPrefix []byte, kc KeyCodec[K], key K) ([]byte, error) {
```

Reply:
i will add a todo, i want to revisit this post schema, maybe we can make the `Prefix` part of schema somehow with a different name